sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/statistic/StatisticSlot.java

```diff
@@ -171,4 +150,16 @@ public void exit(Context context, ResourceWrapper resourceWrapper, int count, Ob
 
         fireExit(context, resourceWrapper, count);
     }
+
+    private void recordCompleteFor(Node node, int batchCount, long rt, Throwable error) {
+        if (node == null) {
+            return;
+        }
+        node.addRtAndSuccess(rt, batchCount);
+        node.decreaseThreadNum();
+
+        if (error != null && !(error instanceof BlockException)) {
+            node.increaseExceptionQps(batchCount);
+        }
+    }
```

Comment:
This `recordCompleteFor` makes it more clear to record for related node. 
In `StatisticSlot#exit` of previous version, there is a judgement for max rt: 
```java
int maxStatisticRt = SentinelConfig.statisticMaxRt();
if (rt > maxStatisticRt) {
    rt = maxStatisticRt;
}
```
Is it still needed here?
I found that in `MetricBucket`
```java
public void addRT(long rt) {
    add(MetricEvent.RT, rt);
    // Not thread-safe, but it's okay.
    if (rt < minRt) {
        minRt = rt;
    }
}
```
If a RT which is greater than default maxRT, it will still be added with actual value,
not quite sure about it, and the minRT seems not the same meaning as judgement of maxRT.

Reply:
> This `recordCompleteFor` makes it more clear to record for related node.
> In `StatisticSlot#exit` of previous version, there is a judgement for max rt:
> 
> ```java
> int maxStatisticRt = SentinelConfig.statisticMaxRt();
> if (rt > maxStatisticRt) {
>     rt = maxStatisticRt;
> }
> ```
> 
> Is it still needed here or some other place?

It's not needed anymore. The maxStatisticRt was designed to avoid affecting the avgRt by extremely-slow requests, which could undermine the effect of RT-based circuit breaking. Now we'll record the original response time (so that the monitoring data could be more accurate). The RT-based circuit breaking strategy will be refactored in the incoming PRs so that it will not suffer from high avgRt.

> I found that in `MetricBucket`
> 
> ```java
> public void addRT(long rt) {
>     add(MetricEvent.RT, rt);
>     // Not thread-safe, but it's okay.
>     if (rt < minRt) {
>         minRt = rt;
>     }
> }
> ```
> 
> If a RT which is greater than default maxRT, it will still be added with actual value,
> not quite sure about it, and the minRT seems not the same meaning as judgement of maxRT.

It doesn't matter for minRt, which is used for adaptive system protection during the BBR period. The maxStatisticRt could be deprecated later.

Reply:
Got it, thanks!